### PR TITLE
Added a new advanced example

### PR DESCRIPTION
**Pull request by Arsh0508**
Although this library had examples for basic functionalities, I have added an advanced example showing several functionalities in a single code.

### CODE REVIEW

**Comment by Arsh0508** [comment]
Also this code can be used to control a robot

File: examples/robot/robot.ino [hunk 1]
```diff
@@ -0,0 +1,302 @@
+//This is an arduino code that can be used to control a robot
+
+//Header inclusions
+#include <LiquidCrystal.h>
+#include <SoftwareSerial.h>
+#include <LedControl.h>
```
Comment (per1234) [inline examples/robot/robot.ino]: Please add a comment directing the user where they can get this library. If it's available from Arduino Library Manager, you can use a special link form that opens a Library Manager search in the Arduino IDE:
```c++
#include <LedControl.h>  // http://librarymanager#LedControl
```
but that doesn't work in the Arduino Web Editor, Arduino Pro IDE, or arduino-cli, so it's necessary to add guidance for those users as well. Maybe something like:
```c++
#include <LedControl.h>  // Install the LedControl library via Library Manager. If using the Arduino IDE, click here: http://librarymanager#LedControl
```

File: examples/robot/robot.ino [hunk 2]
```diff
@@ -0,0 +1,302 @@
+//This is an arduino code that can be used to control a robot
```
Comment (per1234) [inline examples/robot/robot.ino]:
```suggestion
//This is an Arduino sketch that can be used to control a robot
```

File: examples/robot/robot.ino [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+byte b81[] = {B00000,B00000,B00000,B11111,B00000,B00000,B00000,B00000};
+byte b91[] = {B00000,B00000,B00000,B11111,B00000,B00000,B00000,B00000};
+byte b101[] = {B00000,B00000,B00000,B11111,B00000,B00000,B00000,B00000};
+byte b111[] = {B00000,B00000,B00000,B11111,B00000,B00000,B00000,B00000};
+byte b121[] = {B00000,B00000,B00000,B11111,B00000,B00000,B00000,B00000};
+byte b131[] = {B00000,B00000,B00000,B11111,B00000,B00000,B00000,B00000};
+byte b141[] = {B00001,B00010,B00010,B00100,B00110,B01010,B01001,B10001};
+
+//Motor vars
+SoftwareSerial Blue(2,3);
```
Comment (per1234) [inline examples/robot/robot.ino]: You don't appear to use this anywhere in the sketch.

File: examples/robot/robot.ino [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+long int data;
+long int command1 = 92;// forward
+long int command2 = 87; // left
+long int command3 = 65; //  right
+long int command4 = 79; //  Reverse
+long int command5 = 26; // stop
+char state = 0;
+
+//LED vars
+LedControl lc = LedControl(DIN, CLK, CS, 1);
```
Comment (per1234) [inline examples/robot/robot.ino]: You're referencing `DIN`, `CLK`, and `CS` before they are declared.

File: examples/robot/robot.ino [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+    digitalWrite(play_l, LOW);
+
+    digitalWrite(touch, LOW);
+
+}
+
+void loop()
+{
+
+
```
Comment (per1234) [inline examples/robot/robot.ino]:
```suggestion
```
Remove pointless blank lines

File: examples/robot/robot.ino [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+int in1 = 8;
+int in2 = 9;
+int in3 = 10;
+int in4 = 11;
+//LED pins
+int DIN = 13;
+int CS =  12;
+int CLK = 11;
+//Speaker pins
+int play_l = 2;
```
Comment (per1234) [inline examples/robot/robot.ino]: You don't explain what this is. Clearly it's not just a speaker wired to pin 2.

**Comment by Arsh0508** [comment]
Well, I am sorry sir, I am a bit new to open source contributing. But I do promise to not repeat my mistakes and that I will keep learning from them.
Cheers!
Arsh

**Pull request closed without merging**